email-service/src/main/java/com/vgur/spring/email/services/EmailSending.java

```diff
@@ -10,30 +12,59 @@
 import org.springframework.amqp.rabbit.core.RabbitTemplate;
 import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
 import org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.mail.SimpleMailMessage;
 import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+
 @Service
 @RequiredArgsConstructor
 @Slf4j
+@EnableConfigurationProperties(
+        EmailProperties.class
+)
 public class EmailSending {
     private final JavaMailSender mailSender;
+    private final EmailProperties emailProperties;
+    private ArrayList<OrderDto> ordersToSend;
+
+    @PostConstruct
+    public void init(){
+        ordersToSend = new ArrayList<>();
+    }
+
     @Bean
     public Queue emailSendingQueue(){
         return new Queue("EmailSending");
     }
 
     @RabbitListener(queues = "EmailSending")
     public void listen(OrderDto orderDto){
-        sendSimpleMessage(orderDto.getEmail(),"new order","You make new order");
+        ordersToSend.add(orderDto);
     }
+    @Scheduled(fixedDelayString ="${spring.mail.sending_delay}" )
+    private synchronized void sendEmails(){
```

Comment:
why did u use synchronized here ?

Reply:
Probably synchronized will not work here. I need the ordersToSend here to definitely not be changed in another thread. And I don't know how to do it.

Reply:
try to use locks